File: src/main/java/com/temelyan/pomoapp/web/user/AbstractUserController.java
package com.temelyan.pomoapp.web.user;

import com.temelyan.pomoapp.AuthorizedUser;
import com.temelyan.pomoapp.model.User;
import com.temelyan.pomoapp.service.UserService;
import com.temelyan.pomoapp.to.UserTo;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;

import javax.validation.Valid;

import static com.temelyan.pomoapp.Util.ValidationUtil.assureIdConsistent;
import static com.temelyan.pomoapp.Util.ValidationUtil.checkNew;

public class AbstractUserController {
    protected final Logger logger = LoggerFactory.getLogger(getClass());

    @Autowired
    UserService userService;

    protected void create(User user) {
        logger.info("create {}", user);
        checkNew(user);
        userService.create(user);
    }

    protected void update(@Valid UserTo userTo, int id) {
        logger.info("update {} with id={}", userTo, id);
        assureIdConsistent(userTo, AuthorizedUser.id());
        userService.update(userTo);
    }
}

File: src/main/java/com/temelyan/pomoapp/web/user/AbstractUserController.kt
package com.temelyan.pomoapp.web.user

import com.temelyan.pomoapp.model.User
import com.temelyan.pomoapp.service.AuthorisedUserService
import com.temelyan.pomoapp.service.UserService
import com.temelyan.pomoapp.to.UserTo
import com.temelyan.pomoapp.util.ValidationUtil.assureIdConsistent
import com.temelyan.pomoapp.util.ValidationUtil.checkNew
import org.slf4j.LoggerFactory
import javax.validation.Valid

abstract class AbstractUserController(
        var userService: UserService,
        var authorisedUserService: AuthorisedUserService) {
    protected val logger = LoggerFactory.getLogger(javaClass)


    protected fun create(user: User) {
        logger.info("create {}", user)
        checkNew(user)
        userService.create(user)
    }

    protected fun update(@Valid userTo: UserTo, id: Int) {
        logger.info("update {} with id={}", userTo, id)
        assureIdConsistent(userTo, authorisedUserService.get().id())
        val updatedUser = userService.update(userTo)
        authorisedUserService.get().userTo.password = updatedUser.password
    }
}
